Automattic/wp-calypso
# Gutenlypso: apply block availability rules.

#### Changes proposed in this Pull Request

This PR modifies the initialization process of _Gutenlypso_ in such a way that, when a block is not available to the user, it is removed from the _Add Block_ dialog.

| Site on Free Plan | Site on Business Plan | Downgrade Notice |
| - | - | - |
| ![screen shot 2018-12-04 at 23 34 08](https://user-images.githubusercontent.com/233601/49486624-f6040a00-f81d-11e8-87ec-0d6e42386c1b.png) | ![screen shot 2018-12-04 at 23 34 45](https://user-images.githubusercontent.com/233601/49486632-02886280-f81e-11e8-8f49-70acbe512210.png) | ![screen shot 2018-12-04 at 23 55 39](https://user-images.githubusercontent.com/233601/49487216-306ea680-f820-11e8-977b-11472e4a2c6b.png) |

Block availability is retrieved using the `/sites/{siteSlug}/gutenberg/available-extensions` endpoint when the user hits any of the `Gutenberg` routes.

Gutenberg itself is initialized once during the lifetime of Calypso, but every time the user hits one of the Gutenberg routes, `applyGutenbergBlockAvailability` unregisters or registers blocks as specified by the server.

Unlike #28976, this PR makes no changes to the Jetpack code, so no regressions are expected there.

#### Testing instructions



To verify this PR, you'll need two sites, one on the Free or Personal plan, and another on either the Premium or Business Plan.

* Starting on a Site with a Free or Personal plan, Add or Edit a Blog post.
  * If you haven't opt-in to the new editor.
* Add a _Simple Payments Button_ by using the _Add Block_ interface, of by typing `/simple`.
* Without reloading, exit the editor and select a Premium or Business site on _My Sites_
* Add or Edit a blog post. Opt-in to the new editor if you haven't already.
* Try to add a _Simple Payments Button_ by using the _Add Block_ interface or by typing the shorthand.

Simple Payments should work just fine on sites with a Premium or Business plan, but should not appear on the Jetpack list on sites with a Free or Personal plan.

You can also try reloading between sites, and downgrading a site and editing a post with a Simple Payment Button.
